### PR TITLE
Support cdc delete message.

### PR DESCRIPTION
**Pull request by MINCWANG**
Support cdc delete message.

The previous code cannot read the delete message, because the length of the delete message is 0

### CODE REVIEW

File: pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/table/DynamicPulsarDeserializationSchema.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public void deserialize(Message<RowData> message, Collector<RowData> collector)
         outputCollector.inputMessage = message;
         outputCollector.physicalKeyRows = keyCollector.buffer;
         outputCollector.outputCollector = collector;
-        if (message.getData() == null && upsertMode) {
+        if (message.getData() == null || data.length == 0 && upsertMode) {
```
Comment (syhily) [inline pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/table/DynamicPulsarDeserializationSchema.java:141]: `if ((message.getData() == null || data.length == 0) && upsertMode) {`

**Review by syhily — CHANGES_REQUESTED** [review]
The logic should be improved.

**Comment by wolfstudy** [comment]
@syhily Hello, PTAL again thanks.

**Comment by wolfstudy** [comment]
@MINCWANG Can you help to check the action CI?

**Comment by MINCWANG** [comment]
@wolfstudy @syhily  Hi, How do i restart CI? thks

**Comment by syhily** [comment]
> @wolfstudy @syhily Hi, How do i restart CI? thks.. CI have a error for UT

The UT has been restarted. You can restart the UT in `Checks` tab.

**Comment by jianyun8023** [comment]
> > @wolfstudy @syhily Hi, How do i restart CI? thks.. CI have a error for UT
> 
> The UT has been restarted. You can restart the UT in `Checks` tab.

@syhily I'll fix this error. I guess it is caused by the change of MessageId sequence in pulsar 2.7.1.

**Comment by MINCWANG** [comment]
> > > @wolfstudy @syhily Hi, How do i restart CI? thks.. CI have a error for UT
> > 
> > 
> > The UT has been restarted. You can restart the UT in `Checks` tab.
> 
> @syhily I'll fix this error. I guess it is caused by the change of MessageId sequence in pulsar 2.7.1.

@jianyun8023 Thanks, please help me restart CI when you fix this error

**Review by jianyun8023 — APPROVED** [review]

**Review by syhily — APPROVED** [review]

**Pull request merged**